File: Week 2 Arrays/Readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// Declaration
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int compute_index(int letters, int words, int sentences);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Print the numbers of letters, words and sentences
    printf("%i, %i, %i\n", letters, words, sentences);

    // Compute the Coleman-Liau index
    int index = compute_index(letters, words, sentences);
    printf("index = %i\n", index);

    // Print the grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (1 < index && index < 17)
    {
        printf("Grade %i\n", index);
    }
    else if (17 < index)
    {
        printf("Grade 16+\n");
    }
}

// Function to count the number of letters
int count_letters(string text)
{
    int letters = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }

    return letters;
}

// Function to count the number of words

int count_words(string text)
{

    int space = 0;

    // Count the number of space
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            space += 1;
        }
    }

    // the number of words == the number of space + 1
    int words = space + 1;

    return words;
}

// Function to count the number of sentences
int count_sentences(string text)
{
    int sentences = 0;

    // Count the number of punctuation except '
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences += 1;
        }
    }

    return sentences;
}

// Function to compute the Coleman-Liau index
int compute_index(int letters, int words, int sentences)
{
    // Compute L
    float L = (float) letters * 100 / words;
    printf("L = %f\n", L);

    // Compute S
    float S = (float) sentences * 100 / words;
    printf("S = %f\n", S);

    // Compute index
    float index = 0.0588 * L - 0.296 * S - 15.8;
    printf("index = %f\n", index);

    return round(index);
}

File: Week 6 Python/readability.py
def main():

    # Prompt the user for some text
    text = input("Text: ")

    # Count the number of letters, words, and sentences in the text
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Compute the Coleman-Liau index
    index = compute_index(letters, words, sentences)

    # Print the grade level
    if index < 1:
        print("Before Grade 1")
    elif 1 < index and index < 17:
        print(f"Grade {index}")
    elif index > 17:
        print("Grade 16+")


# Function to count the number of letters
def count_letters(text):
    letter_counter = 0
    for i in range(len(text)):
        if str.isalpha(text[i]) == True:
            letter_counter += 1
    return letter_counter


# Function to count the number of words
def count_words(text):
    space_counter = 0
    for i in range(len(text)):
        if str.isspace(text[i]) == True:
            space_counter += 1
    return space_counter + 1  # the number of words == the number of space + 1


# Function to count the number of sentences
def count_sentences(text):
    sentence_counter = 0
    for i in range(len(text)):
        if text[i] in ['.', '!', '?']:
            sentence_counter += 1
    return sentence_counter


# Function to compute the Coleman-Liau index
def compute_index(letters, words, sentences):

    # Compute L
    L = float(letters) * 100 / float(words)

    # Compute S
    S = float(sentences) * 100 / float(words)

    # Compute index
    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)


main()
